File: fsharp/bob/Bob/Bob.fs
module Bob

let hey (msg : string) =
    if not(String.exists(fun c -> not(System.Char.IsWhiteSpace(c))) msg) then "Fine. Be that way!"
    elif (String.exists(fun c -> System.Char.IsLetter(c)) msg)
        && (msg.ToUpper()) = msg then "Whoa, chill out!"
    elif msg.EndsWith("?") then "Sure."
    else "Whatever."

File: python/bob/bob.py
import string


def hey(phrase):
    if (phrase == phrase.upper()) and \
       set(phrase).intersection(string.ascii_uppercase):
        return "Whoa, chill out!"
    elif phrase.split() == []:
        return "Fine. Be that way!"
    elif list(phrase.rstrip())[-1] == '?':
        return "Sure."
    else:
        return "Whatever."
